ens/utils.py

```diff
@@ -81,10 +81,12 @@ def normalize_name(name):
         return name
     elif isinstance(name, (bytes, bytearray)):
         name = name.decode('utf-8')
+    elif name[0:4] == 'xn--':
+        name = idna.decode(name, uts46=True, std3_rules=True)
```

Comment:
I guess this would cause any name encoded as `xn--` to reject the emoji?

Reply:
It would, but the `xn--` prefix is from IDNA encoding AIUI, so the only way to get to an `xn--` prefixed domain name is via `idna.encode` which throws an error if it encounters a character that isn't IDNA compliant (like an emoji).